frontend/src/concepts/pipelines/context/MlmdListContext.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+interface MlmdOrderBy {
+  field: string;
+  direction: 'asc' | 'desc';
+}
+
+interface MlmdListContextProps {
+  filterQuery: string | undefined;
+  pageToken: string | undefined;
+  maxResultSize: number;
+  orderBy: MlmdOrderBy | undefined;
+  setFilterQuery: (filterQuery: string | undefined) => void;
+  setPageToken: (pageToken: string | undefined) => void;
+  setMaxResultSize: (maxResultSize: number) => void;
+  setOrderBy: (orderBy: MlmdOrderBy | undefined) => void;
+}
+
+const MlmdListContext = React.createContext({} as MlmdListContextProps);
+
+export const MlmdListContextProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
+  const [filterQuery, setFilterQuery] = React.useState<string>();
+  const [pageToken, setPageToken] = React.useState<string>();
+  const [maxResultSize, setMaxResultSize] = React.useState(10);
+  const [orderBy, setOrderBy] = React.useState<MlmdOrderBy>();
+  const value = React.useMemo(
+    () => ({
+      filterQuery,
+      pageToken,
+      maxResultSize,
+      orderBy,
+      setFilterQuery,
+      setPageToken,
+      setMaxResultSize,
+      setOrderBy,
+    }),
+    [filterQuery, maxResultSize, orderBy, pageToken],
+  );
+
+  return <MlmdListContext.Provider value={value}>{children}</MlmdListContext.Provider>;
+};
+
+export const useMlmdListContext = (nextPageToken?: string): MlmdListContextProps => {
+  // Force disabled state to pagination when there is no nextPageToken
+  React.useEffect(() => {
+    const paginationNextButtons = document.querySelectorAll('button[aria-label="Go to next page"]');
```

Comment:
The pagination arrow has no exposed disabled handler, and due to the way data is only fetched per page for MLMD resources, this was necessary to disable the next arrow on the last page.

Reply:
can you do something where once you have a next page token, you set the item count to some defiend value. then it will disable on its own 

for ref this is how the page size is calculated
```tsx
 // when itemCount is not known let's set lastPage as page+1 as we don't know the total count
itemCount || itemCount === 0 ? Math.ceil(itemCount / perPage) || 0 : page + 1;
```

Reply:
OK in that case, can you log a PF issue and add that too the code as a comment please. This really should not be a way we do it, but if there is no other option then its fine but should be tracked

Reply:
I created the issue and added a comment with a link to that PF issue.